x-pack/plugins/observability_solution/apm/server/lib/helpers/get_document_sources.ts

```diff
@@ -179,48 +179,29 @@ const getDocumentTypeRequestsFn =
     return rollupIntervals.map((rollupInterval) => ({
       documentType,
       rollupInterval,
-      before: getRequest({
-        documentType,
-        rollupInterval,
-        filters: [...kql, ...beforeRange],
-      }),
-      current: getRequest({
+      documentTypeQuery: getRequest({
         documentType,
         rollupInterval,
         filters: [...kql, ...currentRange],
       }),
-      ...(documentType !== ApmDocumentType.ServiceTransactionMetric
+      ...(isRetroCompatibleDocType(documentType, rollupInterval)
         ? {
-            durationSummaryCheck: getRequest({
+            durationSummaryNotSupportedQuery: getRequest({
               documentType,
               rollupInterval,
-              filters: [...kql, ...currentRange, getDurationLegacyFilter()],
+              filters: [...kql, ...currentRange, isDurationSummaryNotSupportedFilter()],
             }),
           }
-        : {}),
+        : undefined),
     }));
   };
 
-const mapToSources = (sources: DocumentTypeData[], hasAnySourceDocBefore: boolean) => {
-  return sources.map((source) => {
-    const { documentType, hasDocAfter, hasDocBefore, rollupInterval, allHaveDurationSummary } =
-      source;
-
-    const hasDocBeforeOrAfter = hasDocBefore || hasDocAfter;
-
-    // If there is any data before, we require that data is available before
-    // this time range to mark this source as available. If we don't do that,
-    // users that upgrade to a version that starts generating service tx metrics
-    // will see a mostly empty screen for a while after upgrading.
-    // If we only check before, users with a new deployment will use raw transaction
-    // events.
-    const hasDocs = hasAnySourceDocBefore ? hasDocBefore : hasDocBeforeOrAfter;
-
-    return {
-      documentType,
-      rollupInterval,
-      hasDocs,
-      hasDurationSummaryField: allHaveDurationSummary,
-    };
-  });
+const isRetroCompatibleDocType = (
```

Comment:
I'm not sure if I understand the naming of this function?

Reply:
Renamed it. let me know if it's clearer now.

Reply:
works for me!